src/Components/Server/src/DependencyInjection/IServerSideBlazorBuilder.cs

```diff
@@ -1,15 +1,19 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using Microsoft.AspNetCore.Components.Endpoints;
+
 namespace Microsoft.Extensions.DependencyInjection;
 
 /// <summary>
 /// A builder that can be used to configure Server-Side Blazor.
 /// </summary>
-public interface IServerSideBlazorBuilder
+public interface IServerSideBlazorBuilder : IRazorComponentsBuilder
 {
     /// <summary>
     /// Gets the <see cref="IServiceCollection"/>.
     /// </summary>
-    IServiceCollection Services { get; }
+    public new IServiceCollection Services { get; }
```

Comment:
How come this is being made public here *and* there's an explicit interface implementation below? Wouldn't just one of those be sufficient?

Reply:
Perhaps this is to avoid a binary breaking change. I see the base interface has a default implementation of `Services`. AFAIK it would be source-compatible just to add the base interface and remove the property from `IServerSideBlazorBuilder` entirely, but perhaps that's binary-breaking.

Is that the reason or am I missing some point?

Reply:
This is an interface, it was already public.

I had to apply `new` because by making `IServerSideBlazorBuilder` extend `IRazorComponentBuilder` it was hiding the member. If I removed the ServiceCollection from here, it was a breaking change. We can take the breaking change if we want to, but that would impact any library that compiled against `IBlazorServerBuilder` so it just felt safer to do it this way.

Reply:
> Is that the reason or am I missing some point?

We were both typing at the same time. Yes, its to avoid the binary breaking change.